_client/src/Network/ApiManager.cpp

```diff
@@ -193,11 +193,27 @@ void ApiManager::handleRoleList(const Dataset &dataset)
     QVector<Role> roleVector;
     auto idList = dataset[Keys::Role::ID];
     auto nameList = dataset[Keys::Role::NAME];
-    for (auto i = idList.cbegin(), j = nameList.cbegin(); i != idList.cend(); i++, j++)
+    for (auto id = idList.cbegin(), name = nameList.cbegin(); id != idList.cend(); id++, name++)
     {
-        qDebug() << "id - " << *i << ", name - " << *j;
-        Role role(((i)->isEmpty() ? "" : *i), ((j)->isEmpty() ? "" : *j));
-        roleVector.emplace_back(role);
+        Role *role;
+        QString strId = "", strName = "";
+        if (!(id->isEmpty() && name->isEmpty()))
```

Comment:
You call the same method isEmpty() for id and name several times in different if statement. I would recommend call them once before if(), store result in temporal variable and then use it for checks.